src/Microsoft.TemplateEngine.Edge/EngineEnvironmentSettings.cs

```diff
@@ -1,40 +1,55 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#nullable enable
+
 using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Runtime.InteropServices;
 using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge.Settings;
 
-namespace Microsoft.TemplateEngine.Utils
+namespace Microsoft.TemplateEngine.Edge
 {
     public sealed class EngineEnvironmentSettings : IEngineEnvironmentSettings, IDisposable
     {
         private volatile bool _disposed;
 
-        public EngineEnvironmentSettings(ITemplateEngineHost host, Func<IEngineEnvironmentSettings, ISettingsLoader> settingsLoaderFactory)
-            : this(host, settingsLoaderFactory, null)
+        public EngineEnvironmentSettings(
+            ITemplateEngineHost host,
+            bool virtualizeSettings = false,
+            Action<IEngineEnvironmentSettings>? onFirstRun = null,
+            string? settingsLocation = null,
```

Comment:
I wonder if `globalSettingsLocation` would be better name, to clarify this is not HostVersionDir